### PR TITLE
DRAFT: Adds Sqlserver to benchmark/analysis

### PR DESCRIPTION
**Pull request by rothaarlappen**
Adds sql-server tpch-data setup and translated sql scripts

### CODE REVIEW

File: src/database/__init__.py [hunk 1]
```diff
@@ -3,116 +3,123 @@
...
+    db = MASTER_DATABASES[sql_flavor] if scalefactor == 0 else DATABASES[scalefactor]
+    if sql_flavor == "sqlserver":
+        return f"mssql+pyodbc://{SQLSERVER_USER}:{SQLSERVER_PW}@{SQLSERVER_HOST}/{db}?driver=SQL+Server+Native+Client+11.0"
+    else: 
+        return f"postgresql://{PG_USER}:{PG_PW}@{PG_HOST}/{db}"
 
 def create_connection():
-    return create_engine(psycopg2_connection_string())
-
+    print(connectionstring())
```
Comment (Tratori) [inline src/database/__init__.py]: looks like debugging print to me

File: src/datasetup/setup_tpch.py [hunk 2]
```diff
@@ -68,17 +77,44 @@ def setup(scalefactors: List[float], tables: List[str]):
                 + "' with (format csv, delimiter '|');"
             )
             execute_sql(command, db_connstring)
+# TODO (Paul): refactor both setup-scripts and remove redundancy...
+def setup_sqlserver(scalefactors: List[float], tables: List[str]):
+    master_name = "master"
```
Comment (Tratori) [inline src/datasetup/setup_tpch.py]: why call this `master_name`?

**Review by Tratori — APPROVED** [review]

File: src/database/__init__.py (hunk 1 above)
Comment (rothaarlappen) [reply src/database/__init__.py]: you're right..

File: src/evaluation/benchmark.py [hunk 3]
```diff
@@ -124,6 +140,10 @@ def main():
                 time_pipeline_execution(
                     "original", pipeline_file, benchmark_results_pipeline_persist
```
Comment (Tratori) [inline src/evaluation/benchmark.py:142]: I think we should add sql_flavor here as a parameter as well. I don't expect different runtimes, but we should back that up with data...

**Review by Tratori — CHANGES_REQUESTED** [review]

**Review by Tratori — APPROVED** [review]

**Pull request merged**